### PR TITLE
Allow users to opt into review based on linter

### PR DESCRIPTION
**Pull request by salbertson**
This change disabled all linters by default unless the linter is enabled in `.hound.yml` or configured using `config_file`.

This, along with a migration path for existing users, should make Hound configuration more explicit and avoid the case where team members are surprised by what Hound comments on after enabling the service.

### CODE REVIEW

File: app/models/hound_config.rb [hunk 1]
```diff
@@ -1,18 +1,15 @@
...
-    eslint
-    jscs
-    jshint
-    mdast
-    python
-    swift
-  )
-  LANGUAGES = %w(
+  # Only used in Config::Ruby
+  LINTERS = %w(
```
Comment (salbertson) [inline app/models/hound_config.rb]: Do we need an array of supported languages anymore? It's only used in one remaining place, to check for "legacy" config files.

Comment (teoljungberg) [inline app/models/hound_config.rb]: I don't think we need it, so I agree.

**Comment by teoljungberg** [comment]
This is pretty sweet, and I'm a fan of this.

What can we do to move this forward @salbertson ?

**Comment by gylaz** [comment]
Sounds like we aren't making progress on this, and we don't really have a way to support our existing clients (since they expect linters to be on by default). I'm going to close this for now and we can re-visit when we are prepared.

**Pull request closed without merging**